### PR TITLE
Create @woocommerce/admin-layout package

### LINKED ISSUE #36721: Create a @woocommerce/admin-layout package
The header slots (`WooHeaderItem` and `WooHeaderPageTitle`) and footer slots (`WooFooterItem`) should be moved to a new package, `@woocommerce/admin-layout` so that they can be used across packages and within 3rd party plugins.

Eventually, it would be good to move other header and footer components into this package, but for now we can begin with just moving the slots so they can be filled in the product editor.

**Acceptance Criteria**
* [ ] A new package is created with the name `@woocommerce/admin-layout`
* [ ] `WooHeaderItem` is exposed via this package
* [ ] `WooHeaderNavigationItem` is exposed via this package
* [ ] `WooHeaderPageTitle` is exposed via this package
* [ ] `WooFooterItem` is exposed via this package

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This adds a new package: `@woocommerce/admin-layout` with the following initial components moved over from `woocommerce/client/admin`:

* `WooHeaderItem`
* `WooHeaderNavigationItem`
* `WooHeaderPageTitle`
* `WooFooterItem`

More admin layout-specific components will be moved over in the future.

Closes #36721.

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Load this branch and run `pnpm install && pnpm run build` and make sure everything builds correctly.
2. Smoke test WCAdmin pages and make sure that the headers and footers appears correctly.
   
    * Headers on all WCAdmin pages
    * CES footer on new product editor page.

3. Open your console and check if `window.wc.adminLayout` exists with the following components:

* `WC_FOOTER_SLOT_NAME`
* `WC_HEADER_NAVIGATION_SLOT_NAME`
* `WC_HEADER_PAGE_TITLE_SLOT_NAME`
* `WC_HEADER_SLOT_NAME`
* `WooFooterItem`
* `WooHeaderItem`
* `WooHeaderNavigationItem`
* `WooHeaderPageTitle`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/admin-layout/.npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+pacakge-lock=false
```
Comment (nathanss) [inline packages/js/admin-layout/.npmrc]: Typo here!

File: packages/js/admin-layout/src/plugins/woo-footer-item/index.tsx [hunk 2]
```diff
@@ -27,7 +29,10 @@ export const WC_FOOTER_SLOT_NAME = 'woocommerce_footer_item';
  * @param {Array}  param0.children - Node children.
  * @param {Array}  param0.order    - Node order.
  */
-export const WooFooterItem: React.FC< { order?: number } > & {
+export const WooFooterItem: React.FC< {
+	children?: React.ReactNode;
```
Comment (nathanss) [inline packages/js/admin-layout/src/plugins/woo-footer-item/index.tsx:33]: Why did `children` was added here?

**Review by nathanss — CHANGES_REQUESTED** [review]

**Comment by nathanss** [comment]
I couldn't find any issues with smoke tests! All good. Please ping me again when you fix the conflicts and I'll re-review 😄

File: packages/js/admin-layout/src/plugins/woo-footer-item/index.tsx (hunk 2 above)
Comment (mattsherman) [reply packages/js/admin-layout/src/plugins/woo-footer-item/index.tsx:33]: In React 18, the `children` prop is no longer implicitly on `FC`. So, you have to explicitly define it.

https://www.newline.co/@kchan/what-happened-to-the-fc-types-implicit-children-prop-in-typesreact-v18--38576c59

**Comment by mattsherman** [comment]
@nathanss Ready for re-review, though be aware that we may go with `@woocomerce/admin-header` and `@woocommerce/admin-footer`.

**Comment by mattsherman** [comment]
@samueljseay

Output of `pnpm syncpack list-matches`:

```
= Version Group 10 =============================================================
✘ @wordpress/a11y is pinned in this version group at wp-6.0
✘ @wordpress/api-fetch is pinned in this version group at wp-6.0
✘ @wordpress/base-styles is pinned in this version group at wp-6.0
✘ @wordpress/browserslist-config is pinned in this version group at wp-6.0
  ^4.1.1 in devDependencies of packages/js/admin-layout/package.json
✘ @wordpress/components is pinned in this version group at wp-6.0
  ^19.5.0 in dependencies of packages/js/admin-layout/package.json
✘ @wordpress/compose is pinned in this version group at wp-6.0
✘ @wordpress/core-data is pinned in this version group at wp-6.0
✘ @wordpress/data is pinned in this version group at wp-6.0
✘ @wordpress/data-controls is pinned in this version group at wp-6.0
✘ @wordpress/date is pinned in this version group at wp-6.0
✘ @wordpress/deprecated is pinned in this version group at wp-6.0
✘ @wordpress/dom is pinned in this version group at wp-6.0
✘ @wordpress/dom-ready is pinned in this version group at wp-6.0
✘ @wordpress/element is pinned in this version group at wp-6.0
  ^4.1.1 in dependencies of packages/js/admin-layout/package.json
✘ @wordpress/hooks is pinned in this version group at wp-6.0
✘ @wordpress/html-entities is pinned in this version group at wp-6.0
✘ @wordpress/i18n is pinned in this version group at wp-6.0
✘ @wordpress/icons is pinned in this version group at wp-6.0
✘ @wordpress/interface is pinned in this version group at wp-6.0
✘ @wordpress/keyboard-shortcuts is pinned in this version group at wp-6.0
✘ @wordpress/keycodes is pinned in this version group at wp-6.0
✘ @wordpress/media-utils is pinned in this version group at wp-6.0
✘ @wordpress/notices is pinned in this version group at wp-6.0
✘ @wordpress/plugins is pinned in this version group at wp-6.0
✘ @wordpress/primitives is pinned in this version group at wp-6.0
✘ @wordpress/rich-text is pinned in this version group at wp-6.0
✘ @wordpress/url is pinned in this version group at wp-6.0
✘ @wordpress/warning is pinned in this version group at wp-6.0
```

However, running `pnpm syncpack fix-mismatches` does not fix these up (it did fix up `@types/wordpress__components` and `postcss-loader` in fc1dfee8fe17e4569cf4b1275249b79fc947c0b1.

I'm not sure what's going on!

**Comment by JamieMason** [comment]
> However, running `pnpm syncpack fix-mismatches` does not fix these up (it did fix up `@types/wordpress__components` and `postcss-loader` in [fc1dfee](https://github.com/woocommerce/woocommerce/commit/fc1dfee8fe17e4569cf4b1275249b79fc947c0b1).
> 
> I'm not sure what's going on!

Hey @mattsherman, it might not be this but the [long regex in `filter`](https://github.com/woocommerce/woocommerce/blob/4cbb607631caa65df1d9f9401cfa6ec1d92c4771/.syncpackrc#L3) _could_ be the cause as that acts as a global filter which overrides everything else in the config  – using just the [versionGroups](https://jamiemason.github.io/syncpack/config/version-groups) and [`isIgnored`](https://jamiemason.github.io/syncpack/config/version-groups#isignored-boolean) might work out as an easier to read approach to controlling which packages you want to run syncpack on and which you don't.

Docs on `filter`: https://jamiemason.github.io/syncpack/config/filter

**Review by nathanss — APPROVED** [review]

**Comment by mattsherman** [comment]
> > However, running `pnpm syncpack fix-mismatches` does not fix these up (it did fix up `@types/wordpress__components` and `postcss-loader` in [fc1dfee](https://github.com/woocommerce/woocommerce/commit/fc1dfee8fe17e4569cf4b1275249b79fc947c0b1).
> > I'm not sure what's going on!
> 
> Hey @mattsherman, it might not be this but the [long regex in `filter`](https://github.com/woocommerce/woocommerce/blob/4cbb607631caa65df1d9f9401cfa6ec1d92c4771/.syncpackrc#L3) _could_ be the cause as that acts as a global filter which overrides everything else in the config – using just the [versionGroups](https://jamiemason.github.io/syncpack/config/version-groups) and [`isIgnored`](https://jamiemason.github.io/syncpack/config/version-groups#isignored-boolean) might work out as an easier to read approach to controlling which packages you want to run syncpack on and which you don't.
> 
> Docs on `filter`: https://jamiemason.github.io/syncpack/config/filter

@JamieMason Thanks for the tip using `versionGroups` and `isIgnored` (and for syncpack in general!)! I don't *think* the filter is the issue, as syncpack was correctly noting the mismatches with `syncpack list-matches` but not fixing them with `pnpm syncpack fix-mismatches `. Unless the filtering is implemented differently for the listing and the fixing. 

@samueljseay That said, given that we have `versionGroups`, do we really even need the global `filter`?

**Pull request merged**